src/classes/CW_AudienceGroupTriggerHandler.cls

```diff
@@ -0,0 +1,219 @@
+/**
+* File Name :   CW_AudienceGroupTriggerHandler.cls
+* Description : D-106803
+*
+* Modification Log
+* ============================================================================
+* Ver Date	   Author		 Modification
+* --- ---------- -------------- --------------------------
+* 0.1 24/04/2014 AleX Panka      Initial version
+* 0.2 25/04/2014 Klea Kolaric    B-111670
+* 0.3 03/06/2014 Klea Kolaric    B-114503
+*
+*
+*/
+//Cristina P: 25/06/14: Changing the class from "with sharing" to "without sharing" 
+//so the user can create related records while cost/brief is in approval
+public without sharing class CW_AudienceGroupTriggerHandler {
+	
+	public static final String SEPARATOR = '_';
+	
+	public static void OnBeforeUpdateValidation(List<CW_Audience_Group__c> oldRecs, Map<Id,CW_Audience_Group__c> newRecs){
+		List<Id> IdsToCheck= new List<Id>();
+		//As trhere is less control groups we fiter for them
+		for(CW_Audience_Group__c Oldrec :oldRecs ){
+			if(OldRec.Is_Control_Group__c){
+				CW_Audience_Group__c NewRec=newRecs.get(OldRec.ID);
+				if (!NewRec.Is_Control_Group__c){
+					//We have winner ;-)
+					IdsToCheck.Add(OldRec.ID);
+				}
+			}
+		}
+		if( IdsToCheck.Size()>0){
+			system.debug('#############There a records to chceck');
+			for(CW_Audience_Group__c Rec: [SELECT ID, 
+																(Select Id  From Control_Groups__r Limit 1) 
+						  								FROM CW_Audience_Group__c  
+						  								WHERE ID =: IdsToCheck]){
+				if(Rec.Control_Groups__r==null)	
+					continue;
+			}
+		}else system.debug('#############There a no records chceck');
+	}
+	
+	// update AGS if major changes on AG
+	public static void updateAGS(Map<Id,CW_Audience_Group__c> newTriggerMap, Map<Id,CW_Audience_Group__c> oldTriggerMap)
```

Comment:
Style issue: methods should use the "lowerCamelCase" notation. Suggestion: rename accordingly.